python/tests/test_compile.py

```diff
@@ -605,6 +605,14 @@ def fun(x, y):
         with self.assertRaises(ValueError):
             out = fun(mx.array(0.0), y=MyClass())
 
+    def test_compile_create_list(self):
+        @mx.compile
+        def fun():
+            return [0.1 * mx.zeros((2,)), 0.1 * mx.zeros((2,))]
+
+        out = fun()
+        mx.eval(out)
```

Comment:
@angeloskath piggy backing another bug fix here if you want to see. This is actually an issue with simplify merging outputs when it shouldn't be (so the parents map gets invalid which then causes fusion to fuse in cases that it shouldn't). Another difficult one to track down 😅 .

Reply:
Wow! Very nice!